### PR TITLE
Optimize SQLite performance

### PR DESCRIPTION
**Pull request by sdww0**
This PR will optimize the performance of SQLite in ramfs and ext2:
1. SQLite-ramfs: Performance improves from 3.759s to 2.201s, reaching 94% of Linux performance.
2. SQLite-ext2: Performance improves from 4.839s to 3.376s, reaching 97.6% of Linux performance.

### CODE REVIEW

**Comment by sdww0** [comment]
@boterinas rerun

File: kernel/src/syscall/pread64.rs [hunk 1]
```diff
@@ -31,12 +27,15 @@ pub fn sys_pread64(
...
+        let filetable = ctx.process.file_table().lock();
         let mut writer = ctx
             .process
             .root_vmar()
             .vm_space()
             .writer(user_buf_ptr, user_buf_len)?;
-        file.read_at(offset as usize, &mut writer)?
+        filetable
+            .get_file(fd)?
+            .read_at(offset as usize, &mut writer)?
```
Comment (lrh2000) [inline kernel/src/syscall/pread64.rs:38]: This will cause blocking while holding the `file_table` lock. I'm afraid this is undesirable.

Comment (sdww0) [reply kernel/src/syscall/pread64.rs:38]: Ok, I will discard this commit since it had minimal impact on the performance improvement.

**Comment by sdww0** [comment]
@boterinas rerun

**Review by tatetian — APPROVED** [review]
LGMT. Thanks for the contribution!

**Pull request merged**